visualization/dashboard/src/MLIDashboard/NewExplanationDashboard.tsx

```diff
@@ -0,0 +1,390 @@
+            return props.probabilityY[0].length;
+        }
+        if (props.precomputedExplanations && props.precomputedExplanations.localFeatureImportance) {
+            const localImportances = props.precomputedExplanations.localFeatureImportance.scores;
+            if ((localImportances as number[][][]).every(dim1 => {
+                return dim1.every(dim2 => Array.isArray(dim2));
+            })) {
+                return localImportances.length;
+            }
+        }
+        if (props.precomputedExplanations && props.precomputedExplanations.globalFeatureImportance) {
+            // determine if passed in vaules is 1D or 2D
+            if ((props.precomputedExplanations.globalFeatureImportance.scores as number[][])
+                .every(dim1 => Array.isArray(dim1))) {
+                return (props.precomputedExplanations.globalFeatureImportance.scores as number[][]).length;
+            }
+        }
+        // default to regression case
+        return 1;
+    });
+
+    private static buildIndexedNames(length: number, baseString: string): string[] {
+        return Array.from(Array(length).keys())
+        .map(i => localization.formatString(baseString, i.toString()) as string);
+    }
+
+    private static getModelType(props: IExplanationDashboardProps): ModelTypes {
+        // If python gave us a hint, use it
+        if (props.modelInformation.method === "regressor") {
+            return ModelTypes.regression;
+        }
+        switch(NewExplanationDashboard.getClassLength(props)) {
+            case 1:
+                return ModelTypes.regression;
+            case 2:
+                return ModelTypes.binary;
+            default:
+                return ModelTypes.multiclass;
+        }
+    }
+
+    private static buildGlobalProperties(props: IExplanationDashboardProps): IGlobalExplanationProps {
+        const result: IGlobalExplanationProps = {} as IGlobalExplanationProps;
+        if (props.precomputedExplanations &&
+            props.precomputedExplanations.globalFeatureImportance &&
+            props.precomputedExplanations.globalFeatureImportance.scores) {
+            result.isGlobalImportanceDerivedFromLocal = false;
+            if ((props.precomputedExplanations.globalFeatureImportance.scores as number[][])
+                .every(dim1 => Array.isArray(dim1))) {
+                result.globalImportance = (props.precomputedExplanations.globalFeatureImportance.scores as number[][])
+                    .map(classArray => classArray.reduce((a, b) => a + b), 0);
+                result.globalImportanceIntercept = (props.precomputedExplanations.globalFeatureImportance.intercept as number[])
+                    .reduce((a, b) => a + b, 0);
+            } else {
+                result.globalImportance = props.precomputedExplanations.globalFeatureImportance.scores as number[];
+                result.globalImportanceIntercept = props.precomputedExplanations.globalFeatureImportance.intercept as number;
```

Comment:
just a reminder that this still doesn't ever exist on the python side as far as I know

Reply:
its part of the interface contract we defined for the shape of a feature importance explanation. Its optional, and not presented in the UI (because it kept being undefined). But we should build according to the contract